source/core/src/main/com/csse3200/game/services/DialogueBoxService.java

```diff
@@ -67,31 +70,37 @@ public void hideCurrentOverlay() {
     }
 
     /**
-     * Update the current chat overlay if it exists.
+     * Update the current chat overlay if it exists also responsible for unhighlighting an entity if it exists.
      */
     public void updateText(String[][] text) {
         hints = text;
-        if (currentOverlay == null) {
-            // handling if it ever gets deleted when not supposed to
-            currentOverlay = new DialogueBox(hints);
-        } else {
-            currentOverlay.showDialogueBox(text);
+        if (currentEntity != null) {
+            AnimationRenderComponent animator =  currentEntity.getComponent(AnimationRenderComponent.class);
```

Comment:
You don't need to grab the animation render component, you can just call currentEntity.getEvents().trigger(<animationName>) to start the animation.